index.js

```diff
@@ -1,102 +1,113 @@
-        readlineSync.question('\n');
-
-        console.log('6. Deleting block Blob\n');
-        blobService.deleteBlobIfExists(CONTAINER_NAME, BLOCK_BLOB_NAME, function (error) {
-          handleError(error);
-
-          console.log('7. Deleting container\n');
-          blobService.deleteContainerIfExists(CONTAINER_NAME, function (error) {
-            handleError(error);
-
-            fs.unlinkSync(LOCAL_FILE_PATH);
-            fs.unlinkSync(DOWNLOADED_FILE_PATH);
-          });
-        });
-      });
-    });
-  });
-});
-
-function handleError(error) {
-  if (error) {
-    console.error('Exception thrown:\n', error);
-    process.abort();
-  }
+require('dotenv').config();
+
+const path = require('path');
+const args = require('yargs').argv;
+const storage = require('azure-storage');
+
+const blobService = storage.createBlobService();
+const containerName = 'test-container';
+const sourceFilePath = path.resolve('./example.txt');
+const blobName = path.basename(sourceFilePath, path.extname(sourceFilePath));
+
+const createContainer = () => {
+    return new Promise((resolve, reject) => {
+        blobService.createContainerIfNotExists(containerName, { publicAccessLevel: 'blob' }, err => {
+            if(err) {
+                reject(err);
+            } else {
+                resolve({ message: `Container '${containerName}' created` });
+            }
+        });
+    });
+};
+
+const upload = () => {
+    return new Promise((resolve, reject) => {
+        blobService.createBlockBlobFromLocalFile(containerName, blobName, sourceFilePath, err => {
+            if(err) {
+                reject(err);
+            } else {
+                resolve({ message: `Upload of '${blobName}' complete` });
+            }
+        });
+    });
+};
+
+const download = () => {
+    const dowloadFilePath = sourceFilePath.replace('.txt', '.downloaded.txt');
+    return new Promise((resolve, reject) => {
+        blobService.getBlobToLocalFile(containerName, blobName, dowloadFilePath, err => {
+            if(err) {
+                reject(err);
+            } else {
+                resolve({ message: `Download of '${blobName}' complete` });
+            }
+        });
+    });
+};
+
+const list = () => {
+    return new Promise((resolve, reject) => {
+        blobService.listBlobsSegmented(containerName, null, (err, data) => {
+            if(err) {
+                reject(err);
+            } else {
+                resolve({ message: `Items in container '${containerName}':`, data: data });
+            }
+        });
+    });
+};
+
+const deleteBlock = () => {
+    return new Promise((resolve, reject) => {
+        blobService.deleteBlobIfExists(containerName, blobName, err => {
+            if(err) {
+                reject(err);
+            } else {
+                resolve({ message: `Block blob '${blobName}' deleted` });
+            }
+        });
+    });
+};
+
+const _module = {

```

Comment:
The underscore is to differentiate new object from the "module" that exists for exporting, for instance `module.exports`. I chose the singular form because `_module` represents "the module being exported". Does that make sense?

Reply:
Ok